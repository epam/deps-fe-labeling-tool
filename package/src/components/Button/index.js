import { Button, ButtonType } from './Button'
import { IconButton } from './IconButton'
import { TextButton } from './TextButton.styles'

Button.Icon = IconButton

export {
  Button,
  ButtonType,
  TextButton
}
