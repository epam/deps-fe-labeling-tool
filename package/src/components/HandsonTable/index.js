
export * from './models'
export * from './HandsonTable'
