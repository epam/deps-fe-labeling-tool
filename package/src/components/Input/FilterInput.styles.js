import { CloseCircleOutlined } from '@ant-design/icons'
import styled from 'styled-components'

const FilterInputWrapper = styled.div`
  position: relative;
`

const CloseOutlinedIcon = styled(CloseCircleOutlined)`
  color: rgba(0,0,0,0.45);
  position: absolute;
  top: 0.5rem;
  right: 0.7rem;

  &:hover {
    color: rgba(0,0,0,0.65);
  }
`

export {
  FilterInputWrapper,
  CloseOutlinedIcon
}
