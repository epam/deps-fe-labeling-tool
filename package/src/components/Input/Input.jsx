import React from 'react'
import { Input as AntdInput } from 'antd'
import PropTypes from 'prop-types'
import 'antd/lib/input/style/index.less'

const Input = ({ innerRef, ...rest }) => (
  <AntdInput {...rest} ref={innerRef} />
)

Input.TextArea = AntdInput.TextArea

Input.propTypes = {
  innerRef: PropTypes.oneOfType([
    PropTypes.func,
    PropTypes.shape({
      current: PropTypes.instanceOf(Element)
    })
  ])
}

export { Input }
