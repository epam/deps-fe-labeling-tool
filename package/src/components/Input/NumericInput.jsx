import React, { PureComponent } from 'react'
import { InputNumber } from 'antd'
import 'antd/lib/input-number/style/index.less'
import PropTypes from 'prop-types'

const ALLOWED_TO_KEY_DOWN_REGEXP = /^\d+$|Backspace|Delete|ArrowRight|ArrowLeft/

class NumericInput extends PureComponent {
  static propTypes = {
    className: PropTypes.string,
    disabled: PropTypes.bool,
    min: PropTypes.number,
    max: PropTypes.number,
    value: PropTypes.number,
    onChange: PropTypes.func.isRequired,
    formatter: PropTypes.func
  }

  state = {
    value: this.props.value
  }

  componentDidUpdate = (prevProps) => {
    if (prevProps.value !== this.props.value) {
      this.setState({
        value: this.props.value
      })
    }
  }

  onChange = (value) => {
    this.setState({
      value: value
    })

    if (
      value === '' ||
      value == null ||
      Number.isNaN(Number(value)) ||
      (this.props.min !== undefined && value < this.props.min) ||
      (this.props.max !== undefined && value > this.props.max)
    ) {
      return
    }

    this.props.onChange(value)
  }

  onKeyDown = (e) => {
    if (!ALLOWED_TO_KEY_DOWN_REGEXP.test(e.key)) {
      e.preventDefault()
    }
  }

  render = () => (
    <InputNumber
      className={this.props.className}
      min={this.props.min}
      max={this.props.max}
      value={this.state.value}
      onChange={this.onChange}
      disabled={this.props.disabled}
      formatter={this.props.formatter}
      onKeyDown={this.onKeyDown}
    />
  )
}

export {
  NumericInput
}
