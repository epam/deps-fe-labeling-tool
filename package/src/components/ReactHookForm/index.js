export * from './Form'
export * from './FormField'
export * from './FormItem'
export * from './Validators'
export * from './FormFieldType'
