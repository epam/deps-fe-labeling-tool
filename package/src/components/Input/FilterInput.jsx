import React, { useCallback } from 'react'
import PropTypes from 'prop-types'
import { Input } from '@/components/Input'
import {
  FilterInputWrapper,
  CloseOutlinedIcon
} from './FilterInput.styles.js'

const FilterInput = ({
  className,
  placeholder,
  value,
  onChange
}) => {
  const onInputChange = useCallback(
    (e) => onChange({
      ...e,
      value: e.target.value
    }),
    [onChange]
  )

  const onClose = useCallback(
    (e) => onChange({
      ...e,
      value: ''
    }),
    [onChange]
  )

  return (
    <FilterInputWrapper className={className}>
      <Input
        placeholder={placeholder}
        onChange={onInputChange}
        value={value}
      />
      {
        value && (
          <CloseOutlinedIcon
            onClick={onClose}
          />
        )
      }
    </FilterInputWrapper>
  )
}

FilterInput.propTypes = {
  className: PropTypes.string,
  placeholder: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string
}

FilterInput.defaultProps = {
  value: ''
}

export {
  FilterInput
}
