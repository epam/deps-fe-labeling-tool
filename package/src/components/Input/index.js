import { FilterInput } from './FilterInput'
import { Input } from './Input'
import { NumericInput } from './NumericInput'

Input.Numeric = NumericInput
Input.Filter = FilterInput

export {
  Input
}
