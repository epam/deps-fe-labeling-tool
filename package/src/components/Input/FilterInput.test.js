import React from 'react'
import { shallow } from 'enzyme'
import { Input } from '@/components/Input'
import { FilterInput } from './FilterInput'
import { CloseOutlinedIcon } from './FilterInput.styles.js'

describe('Components: FilterInput', () => {
  let defaultProps
  let wrapper

  beforeEach(() => {
    defaultProps = {
      className: 'input',
      placeholder: 'placeholder',
      value: 'filter string',
      onChange: jest.fn()
    }

    wrapper = shallow(<FilterInput {...defaultProps} />)
  })

  it('should render FilterInput with correct props', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should call onChange with correct args on onInputChange', () => {
    wrapper.find(Input).props().onChange({ target: { value: 'foo' } })
    expect(defaultProps.onChange).toHaveBeenCalledWith({ target: { value: 'foo' }, value: 'foo' })
  })

  it('should call onChange on onClose', () => {
    wrapper.find(CloseOutlinedIcon).props().onClick()
    expect(defaultProps.onChange).toHaveBeenCalled()
  })
})
