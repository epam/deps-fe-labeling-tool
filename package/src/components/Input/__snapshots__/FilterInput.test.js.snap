// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: FilterInput should render FilterInput with correct props 1`] = `
<FilterInputstyles__FilterInputWrapper
  className="input"
>
  <Input
    onChange={[Function]}
    placeholder="placeholder"
    value="filter string"
  />
  <FilterInputstyles__CloseOutlinedIcon
    onClick={[Function]}
  />
</FilterInputstyles__FilterInputWrapper>
`;
